tests/testthat/helper-skip.R

```diff
@@ -0,0 +1,22 @@
+# https://github.com/r-lib/testthat/issues/2236
+skip_on_dev_version <- function() {
+  version <- asNamespace(testing_package())[[".__NAMESPACE__."]][["spec"]][[
+    "version"
+  ]] |>
+    package_version() |>
+    unclass() |>
+    getElement(1L)
+
+  if (length(version) < 3 || tail(version, 1L) < 9000) {
+    invisible()
+  } else {
+    skip("Skip on development versions.")
+  }
+}
+
+# Skip on CRAN, but run on R-universe
+skip_on_cran_except_r_universe <- function() {
```

Comment:
Nice!  I didn't realize R-Universe testing doesn't set NOT_CRAN though thinking about it I guess this should have been intuitive.  Nice work with the MY_UNIVERSE flag.

Reply:
Yes, R-universe intentionally provides checks similar to CRAN, so it should only be possible to detect by this environment variable.

Some Rust-based packages I manage detect this environment variable and download binaries from external sources that are not permitted by CRAN.
https://github.com/PRQL/prqlc-r/blob/0cc594f157d64e52905dfa60d5fa10c857f6a4b1/configure#L8-L9